@xen-orchestra/xapi/vdi.mjs

```diff
@@ -153,23 +153,29 @@ class Vdi {
         }
 
         if (nbdClient !== undefined && format === VDI_FORMAT_VHD) {
-          const taskRef = await this.task_create(
+          taskRef = await this.task_create(
             `Exporting content of VDI ${vdiName} using NBD ${changedBlocks !== undefined ? ' and CBT' : ''}`
           )
-          exportStream = stream
           stream = await createNbdVhdStream(nbdClient, exportStream, {
             changedBlocks,
             vdiInfos: { size, uuid, parentUuid: baseParentUuid },
+            onProgress: async progress => {
+              await this.call('task.set_progress', taskRef, progress)
+            },
           })
-          stream.on('progress', progress => this.call('task.set_progress', taskRef, progress))
-          finished(stream, () => {
-            nbdClient.disconnect()
-            exportStream?.destroy() // ensure the source stream is really closed
-
-            this.task_destroy(taskRef).catch(warn)
+          exportStream.on('error', () => {
+            /* no problem here, only the destroy error */
           })
+          exportStream?.destroy() // we won't need it anymore
```

Comment:
I find it weird to destroy this stream when an error occured because it was destroyed.